### PR TITLE
Implement a comvoor like evaluation

### LINKED ISSUE #174: [destab] Implement learning part of the ComVoor-like evaluation

### PR DESCRIPTION
**Pull request by Nitwix**
closing #174 

Depends on https://github.com/HumanLearning2021/HumanLearningApp/pull/191 to be fully functional (including results at the end of evaluation).

Only low level (EvaluationModel) tests done for now. Complete UI tests will be difficult to implement.

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/learning/EvaluationModel.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+/**
+ * Represents the result of an evaluation.
+ * @param successFailureCountPerPhase maps each phase number to a tuple (#successes, #failures)
+ * For example, if the value at index 3 is (17, 13), this means that the learner made 17 successful
+ * sortings, and 13 unsuccessful sortings for the phase that uses 3 categories of the dataset.
+ */
+@Parcelize
+data class EvaluationResult(val successFailureCountPerPhase: List<Pair<Int, Int>>) : Parcelable
+
+
```
Comment (martin-lenweiter) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/learning/EvaluationModel.kt:16]: Some KDoc here would be nice as it is a public class

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/learning/EvaluationModel.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+     * Also checks if the phase or the evaluation is complete
+     */
+    fun addFailure() {
+        nbFailuresInStreak++
+        val (nbSuccesses, nbFailures) = successFailureCountPerPhase[currentPhase]
+        successFailureCountPerPhase[currentPhase] = nbSuccesses to (nbFailures + 1)
+        checkIfPhaseOrEvaluationComplete()
+    }
+
+    private fun checkIfPhaseOrEvaluationComplete() {
```
Comment (martin-lenweiter) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/learning/EvaluationModel.kt:95]: A bit convoluted but I don't see how we could make it simpler.

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/presenter/LearningPresenter.kt [hunk 3]
```diff
@@ -83,6 +83,7 @@ class LearningPresenter(
                     dbMgt.getPicture(rndCatPicIds.random())
                 }
             LearningMode.PRESENTATION -> rndCatRepr
+            LearningMode.EVALUATION -> rndCatRepr // TODO adapt
```
Comment (martin-lenweiter) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/presenter/LearningPresenter.kt:86]: adapt?

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningFragment.kt [hunk 4]
```diff
@@ -37,6 +39,14 @@ class LearningFragment : Fragment() {
 
     lateinit var learningPresenter: LearningPresenter
 
+    /**
+     * model for evaluation mode
+     * A null value represents the fact that the learning mode is not EVALUATION
+     *
+     * TODO ideally the model would be wrapped inside the presenter (but no time right now)
+     */
+    private var evaluationModel: EvaluationModel? = null
```
Comment (martin-lenweiter) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningFragment.kt:48]: Would it be cleaner to use something like an option in scala? (If there's such a thing in kotlin)

**Review by martin-lenweiter — APPROVED** [review]
All good. Just a few remarks/questions but you probably won't have time to address them today so we can merge and look at it next sprint

**Review by NicolasVial — APPROVED** [review]
Nothing to add to Martin's comments. Good job ! Just a little comment about the .xml files. I know you didn't have time to also modify the layouts for tablet and landscape mode but It's not a problem since I think I will do a full cleanup of all those layouts at the end of the project (just for oyu to know).

**Pull request merged**